tidy3d/plugins/dispersion/fit_web.py

```diff
@@ -1,18 +1,58 @@
 """Fit PoleResidue Dispersion models to optical NK data based on web service
 """
 from typing import Tuple, List
-import numpy as np
+from enum import Enum
 import requests
 from pydantic import BaseModel, PositiveInt, NonNegativeFloat, PositiveFloat, Field
 
 from ...components.types import Literal
 from ...components import PoleResidue
-from ...constants import MICROMETER
-from ...log import log
+from ...constants import MICROMETER, HERTZ
+from ...log import log, WebError, Tidy3dError
 from .fit import DispersionFitter
 
 
-class FitterData(BaseModel):
+class AdvancedFitterParam(BaseModel):
+
+    """Advanced fitter parameters"""
+
+    bound_amp: NonNegativeFloat = Field(
+        None,
+        title="Upper bound of oscillator strength",
+        description="Upper bound of oscillator strength in the model "
+        "(The default 'None' will trigger automatic setup based on the "
+        "frequency range of interest).",
+        unis=HERTZ,
+    )
+    bound_f: NonNegativeFloat = Field(
+        None,
+        title="Upper bound of pole frequency",
+        description="Upper bound of pole frequency in the model "
+        "(The default 'None' will trigger automatic setup based on the "
+        "frequency range of interest).",
+        units=HERTZ,
+    )
+    bound_eps_inf: float = Field(
+        1.0,
+        title="Upper bound of epsilon at infinity frequency",
+        description="Upper bound of epsilon at infinity frequency. It must be no less than 1.",
+        ge=1,
+    )
+    constraint: Literal["hard", "soft"] = Field(
+        "hard",
+        title="Type of constraint for stability",
+        description="Stability constraint that is hard but fast to compute, "
+        "or soft but slightly slower to compute.",
```

Comment:
when would user want "soft"?  Could you explain just a little more?  Seems like soft is less stable and slower to compute? but more likely to return a fit?